content/6_Advanced/Treaps.mdx

```diff
@@ -326,6 +328,215 @@ With split and merge, how do we implement the additional operations?
   in lazy propogation). When splitting or merging, we push these tags downwards
   and perform the corresponding operation.
 
+### Implementation
+
+This code solves the above problem, but also provides a generalizable template.
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+// https://judge.yosupo.jp/problem/dynamic_sequence_range_affine_range_sum
+#include <bits/stdc++.h>
+#define int long long
+using namespace std;
+
+// BeginCodeSnip{ModInt class}
+// operations under mod
+struct mint;
+using P = pair<mint, mint>;
+struct mint {
+	const static int M = 998244353;
+	int v = 0;
+	mint() {}
+	mint(int v) { this->v = (v % M + M) % M; }
+	mint operator+(const mint &o) const { return v + o.v; }
+	mint &operator+=(const mint &o) {
+		v = (v + o.v) % M;
+		return *this;
+	}
+	mint operator*(const mint &o) const { return v * o.v; }
+	mint operator-(const mint &o) const { return v - o.v; }
+	mint &operator-=(const mint &o) {
+		mint t = v - o.v;
+		v = t.v;
+		return *this;
+	}
+	mint operator^(int y) const {
+		mint r = 1, x = v;
+		for (y <<= 1; y >>= 1; x = x * x)
+			if (y & 1) r = r * x;
+		return r;
+	}
+	mint inv() const {
+		assert(v);
+		return *this ^ M - 2;
+	}
+	friend istream &operator>>(istream &s, mint &v) {
+		s >> v.v;
+		return s;
+	}
+	friend ostream &operator<<(ostream &s, const mint &v) {
+		s << v.v;
+		return s;
```

Comment:
```suggestion
		return s << v.v;
```just sayin

Reply:
oh shoot i did not know you could do that